dvc/repo/remove.py

```diff
@@ -11,9 +11,16 @@
 
 @locked
 def remove(self: "Repo", target: str, outs: bool = False):
-    stages = self.stage.from_target(target)
+    stages_info = self.stage.collect_granular(target)
 
-    for stage in stages:
-        stage.remove(remove_outs=outs, force=outs)
+    for stage_info in stages_info:
+        stage = stage_info.stage
+        filter_info = stage_info.filter_info
+        if filter_info is not None:
+            # target is a specific output file
+            for out in stage.filter_outs(filter_info):
+                out.remove(True)
```

Comment:
This might not support granularity (eg: when the output is `data` and the target is `data/images/`).

Reply:
I am not sure if we need to support this though.

Reply:
I assume that setting a directory (i.e. `data/images`) as target might be out of the scope as the issue is currently described (We could extend this P.R. to support directories as target, if needed?) but I actually did not test to set output `data` and target something like `data/images/image_0.png`; I will add a test case for that.

Reply:
It seems that setting `outs=["foo"]` and running `dvc.remove("foo/file1"`) is removing all files under `foo`, not only the given target. I added a test for that and I'm reviewing the code

Reply:
@skshetry is the failing test case (which i have just added) supposed to pass, given the feature description?:

```
def test_remove_file_in_subdir_as_target(tmp_dir, scm, dvc):
    dvc.run(
        name="my",
        cmd='mkdir -p "foo" && echo "hello" > foo/file1 && echo "hello" > foo/file2',
        deps=[],
        outs=["foo"],
    )

    dvc.remove("foo/file1")
    # Fails here because `foo` is being completely removed, not just the file specified
    assert "/foo" in  get_gitignore_content()
    dvc.remove("foo/file2")
    assert not (tmp_dir / ".gitignore").exists()
```
Do we want to add support to that?

Reply:
@daavoo, that's expected, as we have not implemented the granular remove logic. Let's not implement this PR, let's discuss it in a separate issue.

To give you a more context, we have this implemented on the checkout. 

https://github.com/iterative/dvc/blob/c63457746332762ef6f203d2e71d8f7f4bde497a/dvc/repo/checkout.py#L113

Which we pass to Stage's checkout and then to the `Output`'s checkout:
https://github.com/iterative/dvc/blob/c63457746332762ef6f203d2e71d8f7f4bde497a/dvc/stage/__init__.py#L531

https://github.com/iterative/dvc/blob/c63457746332762ef6f203d2e71d8f7f4bde497a/dvc/output/base.py#L404-L413

So the support should be similarly implemented in the `Output` and this `filter_info` should be passed along.

Reply:
I am not sure how much this feature of granular remove makes sense.
Do the users expect this complete removal? Or, should we implement the granular remove, or should we forbid it?

Though forbidding it might not be a good idea for consistency. 🤷🏼

Reply:
> @daavoo, that's expected, as we have not implemented the granular remove logic. Let's not implement this PR, let's discuss it in a separate issue.

Sorry for my lack of understanding :sweat_smile: but by `Let's not implement this PR` you mean that:

A) We should close this P.R. and open a separate issue for discussing this feature and "granular remove"
B) We should leave this P.R. as it is, with the current implementation, and also open a separate issue for further discussion on extending the functionality?
C) None of the above  :sweat_smile:

Reply:
@daavoo, sorry for the typo. We can merge this and then implement granular remove later. Otherwise,  the  PR will get large and hard to review.

Reply:
#5772